sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -77,7 +78,7 @@ class BearerTokenCredentialPolicy(_BearerTokenCredentialPolicyBase, HTTPPolicy):
     :raises: :class:`~azure.core.exceptions.ServiceRequestError`
     """
 
-    def on_request(self, request: "PipelineRequest") -> None:
+    def on_request(self, request: PipelineRequest[HTTPRequestType]) -> None:
```

Comment:
You have to inject `HttpPolicy` with HttpRequestType as well to make the connection